<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Client</class>
 <widget class="QWidget" name="Client">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>501</width>
    <height>476</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Tcp Client</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="1">
    <widget class="QLineEdit" name="ipaddr_line_edit">
     <property name="text">
      <string>localhost</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
   </item>
   <item row="0" column="6">
    <widget class="QPushButton" name="status_Change">
     <property name="minimumSize">
      <size>
       <width>24</width>
       <height>24</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>24</width>
       <height>24</height>
      </size>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset resource="img.qrc">
       <normaloff>:/play</normaloff>:/play</iconset>
     </property>
    </widget>
   </item>
   <item row="0" column="0">
    <widget class="QLabel" name="ip_label">
     <property name="text">
      <string>ip:</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
     </property>
    </widget>
   </item>
   <item row="0" column="2">
    <widget class="QLabel" name="port_label">
     <property name="text">
      <string>port:</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
     </property>
    </widget>
   </item>
   <item row="0" column="3">
    <widget class="QSpinBox" name="port">
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="readOnly">
      <bool>false</bool>
     </property>
     <property name="buttonSymbols">
      <enum>QAbstractSpinBox::UpDownArrows</enum>
     </property>
     <property name="maximum">
      <number>65535</number>
     </property>
    </widget>
   </item>
   <item row="0" column="5">
    <widget class="QLineEdit" name="status_line_edit">
     <property name="text">
      <string>unknow</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="1" column="0" colspan="8">
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Receive:</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="title">
         <string/>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QRadioButton" name="radioButton_ascii_receive">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="text">
            <string>ASCII</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="radioButton_latin1_receive">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="text">
            <string>Latin1</string>
           </property>
           <property name="checked">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="radioButton_UTF8_receive">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="text">
            <string>UTF-8</string>
           </property>
           <property name="checked">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="radioButton_hex_receive">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>HEX</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QTextEdit" name="data_textEdit_receive">
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="clear_receive">
        <property name="maximumSize">
         <size>
          <width>100</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="text">
         <string>clear</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="0" colspan="8">
    <widget class="QGroupBox" name="groupBox_3">
     <property name="title">
      <string>Send:</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QGroupBox" name="groupBox_5">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="title">
         <string/>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QRadioButton" name="radioButton_rn">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="text">
            <string>\r\n</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="radioButton_n">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="text">
            <string>\n</string>
           </property>
           <property name="checked">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="radioButton_none">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="text">
            <string>none</string>
           </property>
           <property name="checked">
            <bool>false</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_4">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="title">
         <string/>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QRadioButton" name="radioButton_ascii_send">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="text">
            <string>ASCII</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="radioButton_latin1_send">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="text">
            <string>Latin1</string>
           </property>
           <property name="checked">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="radioButton_UTF8_send">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="text">
            <string>UTF-8</string>
           </property>
           <property name="checked">
            <bool>false</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QTextEdit" name="data_textEdit_send">
        <property name="readOnly">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <item>
         <widget class="QPushButton" name="clear_send">
          <property name="maximumSize">
           <size>
            <width>100</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>clear</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="send">
          <property name="maximumSize">
           <size>
            <width>100</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>send</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item row="0" column="4">
    <widget class="QLabel" name="label">
     <property name="text">
      <string>status:</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
     </property>
    </widget>
   </item>
  </layout>
  <action name="actionStatusChange">
   <property name="text">
    <string>StatusChange</string>
   </property>
  </action>
  <action name="actionSendData">
   <property name="text">
    <string>SendData</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="img.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>status_Change</sender>
   <signal>clicked()</signal>
   <receiver>actionStatusChange</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>473</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>clear_receive</sender>
   <signal>clicked()</signal>
   <receiver>data_textEdit_receive</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>76</x>
     <y>228</y>
    </hint>
    <hint type="destinationlabel">
     <x>119</x>
     <y>175</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>clear_send</sender>
   <signal>clicked()</signal>
   <receiver>data_textEdit_send</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>150</x>
     <y>441</y>
    </hint>
    <hint type="destinationlabel">
     <x>168</x>
     <y>406</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>send</sender>
   <signal>clicked()</signal>
   <receiver>actionSendData</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>345</x>
     <y>444</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
